include/OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.hpp

```diff
@@ -0,0 +1,187 @@
+    /// @param              [in] aPropulsionSystem A propulsion system
+    /// @return             A reference to output stream
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const PropulsionSystem& aPropulsionSystem);
+
+    /// @brief              Check if propulsion system is defined
+    ///
+    /// @return             True if propulsion system is defined
+
+    bool isDefined() const;
+
+    /// @brief              Print propulsion system
+    ///
+    /// @param              [in] anOutputStream An output stream
+    /// @param              [in] (optional) displayDecorators If true, display decorators
+
+    void print(std::ostream& anOutputStream, bool displayDecorator = true) const;
+
+    /// @brief              Get propulsion system's thrust
+    ///
+    /// @code
+    ///                     Scalar thrust = propulsionSystem.getThrust() ;
+    /// @endcode
+    ///
+    /// @return             Scalar
+
+    Scalar getThrust() const;
+
+    /// @brief              Get propulsion system's specific impulse
+    ///                     https://en.wikipedia.org/wiki/Specific_impulse
+    ///
+    /// @code
+    ///                     Scalar specificImpulse = propulsionSystem.getSpecificImpulse() ;
+    /// @endcode
+    ///
+    /// @return             Scalar
+
+    Scalar getSpecificImpulse() const;
+
+    /// @brief              Get propulsion system's mass flow rate
+    ///
+    /// @code
+    ///                     Scalar massFlowRate = propulsionSystem.getMassFlowRate() ;
+    /// @endcode
+    ///
+    /// @return             Scalar
+
+    Scalar getMassFlowRate() const;
+
+    /// @brief              Get propulsion system's acceleration
+    ///
+    /// @code
+    ///                     Real acceleration = propulsionSystem.getAcceleration() ;
+    /// @endcode
+    ///
+    /// @return             Real
+
+    Real getAcceleration(const Mass& aMass) const; // TBI: Should return acceleration unit
+
+    static PropulsionSystem Undefined();
+
+   private:
+    Unit thrustSIUnit_ = Unit::Derived(Derived::Unit(
+        Length::Unit::Meter,
+        {1},
+        Mass::Unit::Kilogram,
+        {1},
+        Time::Unit::Second,
+        {-2},
+        ElectricCurrent::Unit::Undefined,
+        {0},
+        Angle::Unit::Undefined,
+        {0}
+    ));
+    Unit specificImpulseSIUnit_ = Unit::Derived(Derived::Unit(
+        Length::Unit::Undefined,
+        {0},
+        Mass::Unit::Undefined,
+        {0},
+        Time::Unit::Second,
+        {1},
+        ElectricCurrent::Unit::Undefined,
+        {0},
+        Angle::Unit::Undefined,
+        {0}
+    ));
+    Unit massFlowRateSIUnit_ = Unit::Derived(Derived::Unit(
+        Length::Unit::Undefined,
+        {0},
+        Mass::Unit::Kilogram,
+        {1},
+        Time::Unit::Second,
+        {-1},
+        ElectricCurrent::Unit::Undefined,
+        {0},
+        Angle::Unit::Undefined,
+        {0}
+    ));
+
```

Comment:
```suggestion
    Scalar massFlowRate_ = Scalar::Undefined();  /// [kg/s]
```
I would store this for performance.